src/utils.js

```diff
@@ -77,37 +76,52 @@ const writeReportFile = (outputFileName, packageList, customHeaderFileName) => {
   console.info(`${outputFileName}.md created!`);
 };
 
-const getAllCorrectLicenses = (licenseRegex) =>
-  licenses.reduce((acc, correctLicense) => {
-    if (licenseRegex.test(correctLicense)) return [...acc, correctLicense];
-
-    return acc;
-  }, []).flat();
-
-const isValidLicense = (license) => licenses.includes(license);
-
-const transformLicense = (license) => {
-  if (isValidLicense(license)) return license;
-
-  const licenseRegex = new RegExp(license, 'i');
+/**
+ * Parses failOn arguments distinguishing between plain string and
+ * regex like strings
+ *
+ * @param {string[]} args - List of arguments to parse
+ * @returns {{invalid: (string|RegExp)[], valid: (string|RegExp)[]}} - List of valid
+ * and invalid parsed arguments. The invalid object will include unsuccessful parsed regex
+ */
+const parseFailOnArgs = args => args.reduce((total, arg) => {
+  try {
+    const pattern = /^\/(?<pattern>.+)\/$/.exec(arg)?.groups?.pattern;
+    return { ...total, valid: [...total.valid, pattern ? new RegExp(pattern) : arg] };
+  } catch (e) {
+    return { ...total, invalid: [...total.invalid, arg] };
+  }
+}, { valid: [], invalid: [] });
 
-  return license + ',' + getAllCorrectLicenses(licenseRegex);
+/**
+ * Tests if the license argument is equal to or matches the expression argument
+ *
+ * @param {string} license - License to check
+ * @param {string|RegExp} expression - The expression the license will be tested against
+ * @returns {boolean} - Result of the test
+ */
+const licenseMatchesExpression = (license, expression) => {
+  const check = {
+    string: () => license === expression,
+    object: () => expression.test(license)
+  };
+  const typeOfExpression = typeof expression;
```

Comment:
We can also check if it's a RegEx by running `expression instanceof RegExp`. What do you think is better?

Reply:
The idea was to be able to parse any type of arguments with the strategy, but it might not make sense since this function should take in only strings and RegExp's. I think your proposal is much cleaner that this approach taking into consideration my previous statement. Thanks for your feedback